CHT/HeatFlux.C

```diff
@@ -103,6 +103,21 @@ void preciceAdapter::CHT::HeatFlux::read(double* buffer, const unsigned int dim)
     }
 }
 
+bool preciceAdapter::CHT::HeatFlux::isLocationTypeSupported(const bool meshConnectivity) const
+{
+    // We don't distinguish between reading and writing. Thus, we just return true
+    // in case there was connectivity defined
+    if (meshConnectivity)
+        return true;
```

Comment:
This sounds a bit complicated to me. Does this mean that we support:

- face centers if mesh connectivity is defined,
- face nodes if mesh connectivity is defined,
- face centers if mesh connectivity is not defined,

both for reading and writing? If not, does a later/previous error catch this?

What did we have before?

Some documentation would be helpful.

Reply:
We support for writing

- face nodes if mesh connectivity is defined
- face centers if no connectivity is defined

for reading
- face centers

However, we do not distinguish between reading and writing. Thus the function is an `||` of these options.

Reply:
> (Does this also apply to all other modules?)

Unfortunately not,

Reply:
Then let's just write something like

```suggestion
    // For cases with mesh connectivity, we support:
    // - face centers, both for reading and writing
    // - face nodes, only for writing
    // However, since we do not distinguish between reading and writing in the code, we
    // always return true and offload the handling to the user.
    if (meshConnectivity)
        return true;
```

I still think that it is not clear what we are trying to say here.

I also have the feeling that we have a larger issue here, which we are not handling properly. Overall, we have a cross product of:

- module (at least CHT and FSI support some operations on some data on either centers or nodes)
- specific coupling data user (heat flux, displacement, force, ...)
- face centers, face nodes
- reading, writing

I understand that we are still catching some of the combinations which are not meant to work, but not all of them. This can create a false impression to the user that they can rely on error handling to choose a working combination.

This PR is definitely something we need, I am just a bit worried seeing the whole design shaking as we cannot clearly and completely define the checks in a simple way.

(by the way, we should have braces around these one-line branches, for maintainability)

Reply:
Ok, I will add a correct version of your comment in all data modules where we support mesh connectivity.